core-libs/setup/ssr/error-handling/server-error-response-transformers/cms-page-not-found-server-error-response-transformer.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ServerErrorResponseTransformer } from './server-error-response-transformers';
+import { CmsPageNotFoundServerError, CxServerError } from '../server-errors';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Priority } from '@spartacus/core';
+
+/**
+ * A transformer responsible for transforming an HTTP error response into a {@link CmsPageNotFoundServerError}.
+ * The transformer is applicable when the error the URL contains 'cms/pages'.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class CmsPageNotFoundServerErrorResponseTransformer
+  implements ServerErrorResponseTransformer
+{
+  hasMatch(error: unknown): boolean {
+    return (
+      this.isHttpErrorResponse(error) && !!error.url?.includes('cms/pages')
```

Comment:
Now any endpoint that contains `cms/pages` is taken into account, which is prone to mistakes, e.g. `/some/custom/api/cms/pagesStatus` (I made up the example)
What about comparing with precise fully built CMS semantic 'pages' endpoint?
```ts
const pagesEndpoint = this.endpointService.buildUrl('pages');
return ...
  ... (error.url ?? '').startsWith(pagesEndpoint)
```
?

Reply:
Good point. I had a gut feeling the check I've proposed may be error-prone. Thank you for the suggestion.